### PR TITLE
Feat/add changesets

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?
It adds [changesets](https://github.com/changesets/changesets), along with a github workflow file to automate publishing to npm with changesets.

## Why?
It will hopefully standardise and reduce friction from the process of publishing updates to `discussion-rendering`, in line with DevX's [recommendations](https://github.com/guardian/recommendations/blob/main/npm-packages.md#continuous-delivery). Closes [this issue](https://github.com/guardian/dotcom-rendering/issues/4864).

## Todo

- [x] There's some kind of yarn/npm thing [preventing the Compressed Size action from running](https://github.com/guardian/discussion-rendering/runs/8013083434?check_suite_focus=true#step:4:19). Pretty sure this is a known issue (just need to update yarn.lock?) but I can't remember the details right now.
- [ ] Test the workflow
- [ ] Document the release process for contributors to the repo
- [ ] Install the changesets-bot in this repo.

Co-authored by: Max Duval <max.duval@theguardian.com>
Co-authored by: George B <705427+georgeblahblah@users.noreply.github.com>

### CODE REVIEW

**Comment by bryophyta** [comment]
@joecowton1 & @sndrs -- we looked through the documentation for changesets but found some of it a bit unclear. We've largely followed the implementation used in `atoms-rendering` [config](https://github.com/guardian/atoms-rendering/blob/main/.changeset/config.json) and [`workflow.yml`](https://github.com/guardian/atoms-rendering/blob/main/.github/workflows/changesets.yml), with some adaptations to our case.

Given that (iirc) you've both worked with it before, would you mind casting an eye over this draft and see if any issues or omissions jump out? Thanks in advance!

**Comment by joecowton1** [comment]
> @joecowton1 & @sndrs -- we looked through the documentation for changesets but found some of it a bit unclear. We've largely followed the implementation used in `atoms-rendering` [config](https://github.com/guardian/atoms-rendering/blob/main/.changeset/config.json) and [`workflow.yml`](https://github.com/guardian/atoms-rendering/blob/main/.github/workflows/changesets.yml), with some adaptations to our case.
> 
> Given that (iirc from our original conversations about this a couple of months ago) you've both worked with it before, would you mind casting an eye over this draft and see if any issues or omissions jump out? Thanks in advance!

Looks good, you might be able to get away without specifying this? Remember that being an odd one when we setup `atoms-rendering`. Otherwise I'd say its just a case of merging and seeing what goes wrong 👍 
```
   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }} 
```

File: .changeset/config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@2.1.0/schema.json",
+  "changelog": "@changesets/changelog-git",
+  "commit": true,
```
Comment (joecowton1) [inline .changeset/config.json:4]: Might want to remove this in light of issues with auto generated commits: https://github.com/guardian/source/pull/1521

File: .github/workflows/changesets.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+              id: changesets
+              uses: changesets/action@v1
+              with:
+                  publish: yarn release:changesets
+                  title: 'Bump `@guardian/discussion-rendering` version'
+                  commit: 'Bump version'
+              env:
+                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+                  NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (joecowton1) [inline .github/workflows/changesets.yml:34]: Not sure we need this - might be worth trying without first?

**Review by joecowton1 — APPROVED** [review]

**Pull request merged**